src/util.js

```diff
@@ -590,12 +590,22 @@ function getNodeKindIncludingLogical(node) {
  * See http://php.net/manual/en/language.types.string.php#language.types.string.syntax.double
  */
 function useDoubleQuote(node, options) {
-  if (node.isDoubleQuote && options.singleQuote) {
-    const rawValue = node.raw.slice(node.raw[0] === "b" ? 2 : 1, -1);
+  if (node.isDoubleQuote) {
+    if (options.singleQuote) {
+      const rawValue = node.raw.slice(node.raw[0] === "b" ? 2 : 1, -1);
 
-    return rawValue.match(
-      /\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"/
-    );
+      return rawValue.match(
+        /\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"|\$/
+      );
+    }
+  } else {
+    if (!options.singleQuote) {
+      const rawValue = node.raw.slice(node.raw[0] === "b" ? 2 : 1, -1);
+
+      return !rawValue.match(
+        /\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"|\$/
+      );
+    }
```

Comment:
A little shorter (but potentially also more confusing?) way to write this might be:
```js
if (node.isDoubleQuote === options.singleQuote) {
  const rawValue = node.raw.slice(node.raw[0] === "b" ? 2 : 1, -1);
  const isComplex = !rawValue.match(
    /\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"|\$/
  );
  return node.isDoubleQuote ? isComplex : !isComplex;
}
```

Reply:
It makes me sad that I didn't automatically think of this.